In DSpace 8, no new item can be deposited

**Describe the bug**
I used the sandbox and submit with Demo Submitter account in the collection "New collection created as Site Administrator" (https://sandbox.dspace.org/handle/10673/1159). I add an item using the Pubmed Import feature of DSpace. I am able to save the item for later, but each time I try to deposit the item, I get an error:   "There was an error when submitting the item, please try again." 

Can save/save for later and find the submission in "My DSPace."

**To Reproduce**
Steps to reproduce the behavior:
1. As submitter, submit a new item in a collection (Entity type Publication)
2. Fullfill all mandatory metadata, add a file and aproove licence
3. Clic on Deposit, 
4. The error displays on screen and the item cannot be deposited.

I've dug into this. 
It appears to be an issue with sending emails from DSpace 8 after the backend was upgraded to Jakarta EE in https://github.com/DSpace/DSpace/pull/9321.  The error occurs in *sending* an email notification, and it appears to block the submission from completing.  Here's the error in the dspace.log:
```
2024-04-11 12:25:54,727 ERROR 541bfbc1-43a4-425c-9eda-66f76bfa7d05 b13c8585-952a-4083-8978-9f9e03b241e2 org.dspace.app.rest.exception.DSpaceApiExceptionControllerAdvice @ An exception has occurred (status:500)
 jakarta.servlet.ServletException: Handler dispatch failed: java.lang.IllegalAccessError: failed to access class com.sun.activation.registries.LogSupport from class jakarta.activation.MailcapCommandMap (com.sun.activation.registries.LogSupport and jakarta.activation.MailcapCommandMap are in unnamed module of loader org.apache.catalina.loader.ParallelWebappClassLoader @5b6ec132)
        at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1104) ~[spring-webmvc-6.1.5.jar:6.1.5]
...
Caused by: java.lang.IllegalAccessError: failed to access class com.sun.activation.registries.LogSupport from class jakarta.activation.MailcapCommandMap (com.sun.activation.registries.LogSupport and jakarta.activation.MailcapCommandMap are in unnamed module of loader org.apache.catalina.loader.ParallelWebappClassLoader @5b6ec132)
        at jakarta.activation.MailcapCommandMap.<init>(MailcapCommandMap.java:149) ~[jakarta.activation-2.0.1.jar:2.0.1]
        at jakarta.activation.CommandMap.getDefaultCommandMap(CommandMap.java:55) ~[jakarta.activation-2.0.1.jar:2.0.1]
        at jakarta.activation.DataHandler.getCommandMap(DataHandler.java:127) ~[jakarta.activation-2.0.1.jar:2.0.1]
        at jakarta.activation.DataHandler.getDataContentHandler(DataHandler.java:589) ~[jakarta.activation-2.0.1.jar:2.0.1]
        at jakarta.activation.DataHandler.writeTo(DataHandler.java:289) ~[jakarta.activation-2.0.1.jar:2.0.1]
        at jakarta.mail.internet.MimeUtility.getEncoding(MimeUtility.java:316) ~[jakarta.mail-2.0.1.jar:2.0.1]
        at jakarta.mail.internet.MimeBodyPart.updateHeaders(MimeBodyPart.java:1551) ~[jakarta.mail-2.0.1.jar:2.0.1]
        at jakarta.mail.internet.MimeMessage.updateHeaders(MimeMessage.java:2246) ~[jakarta.mail-2.0.1.jar:2.0.1]
        at jakarta.mail.internet.MimeMessage.saveChanges(MimeMessage.java:2206) ~[jakarta.mail-2.0.1.jar:2.0.1]
        at jakarta.mail.Transport.send(Transport.java:99) ~[jakarta.mail-2.0.1.jar:2.0.1]
        at org.dspace.core.Email.send(Email.java:482) ~[dspace-api-8.0-SNAPSHOT.jar:8.0-SNAPSHOT]
```

This was likely not found previously because many developers will *disable* email sending on local dev machines.  I'm looking into a possible fix.  It's likely a minor dependency error.

--- a/dspace-api/pom.xml
+++ b/dspace-api/pom.xml
@@ -458,7 +458,12 @@
             <artifactId>commons-validator</artifactId>
         </dependency>
         <dependency>
-            <groupId>com.sun.mail</groupId>
+            <groupId>jakarta.mail</groupId>
+            <artifactId>jakarta.mail-api</artifactId>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.eclipse.angus</groupId>
             <artifactId>jakarta.mail</artifactId>
         </dependency>
         <dependency>

--- a/dspace-services/pom.xml
+++ b/dspace-services/pom.xml
@@ -102,7 +102,12 @@
             <scope>provided</scope>
         </dependency>
         <dependency>
-            <groupId>com.sun.mail</groupId>
+            <groupId>jakarta.mail</groupId>
+            <artifactId>jakarta.mail-api</artifactId>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.eclipse.angus</groupId>
             <artifactId>jakarta.mail</artifactId>
         </dependency>
         <dependency>

--- a/pom.xml
+++ b/pom.xml
@@ -34,8 +34,8 @@
         <jackson.version>2.16.0</jackson.version>
         <jackson-databind.version>2.16.0</jackson-databind.version>
         <jakarta-annotation.version>2.1.1</jakarta-annotation.version>
-        <jaxb-api.version>4.0.0</jaxb-api.version>
-        <jaxb-runtime.version>4.0.1</jaxb-runtime.version>
+        <jaxb-api.version>4.0.2</jaxb-api.version>
+        <jaxb-runtime.version>4.0.5</jaxb-runtime.version>
         <jcache-version>1.1.1</jcache-version>
         <!-- NOTE: Jetty needed for Solr, Handle Server & tests -->
         <jetty.version>9.4.54.v20240208</jetty.version>
@@ -1522,9 +1522,22 @@
                 <version>2.12.5</version>
             </dependency>
             <dependency>
-                <groupId>com.sun.mail</groupId>
+                <groupId>jakarta.activation</groupId>
+                <artifactId>jakarta.activation-api</artifactId>
+                <version>2.1.3</version>
+            </dependency>
+            <!-- Jakarta mail API definition -->
+            <dependency>
+                <groupId>jakarta.mail</groupId>
+                <artifactId>jakarta.mail-api</artifactId>
+                <version>2.1.3</version>
+                <scope>provided</scope>
+            </dependency>
+            <!-- Jakarta mail default implementation/provider -->
+            <dependency>
+                <groupId>org.eclipse.angus</groupId>
                 <artifactId>jakarta.mail</artifactId>
-                <version>2.0.1</version>
+                <version>2.0.3</version>
             </dependency>
             <dependency>
                 <groupId>jakarta.servlet</groupId>

